tests/QueryCompatibleInputMiddlewareTest.php

```diff
@@ -0,0 +1,256 @@
+<?php
+namespace Aws\Test;
+
+use Aws\AwsClient;
+use Aws\Command;
+use Aws\HandlerList;
+use Aws\Api\Service;
+use Aws\Middleware;
+use Aws\QueryCompatibleInputMiddleware;
+use GuzzleHttp\Psr7\Request;
+use Yoast\PHPUnitPolyfills\TestCases\TestCase;
+
+/**
+ * @covers \Aws\QueryCompatibleInputMiddleware
+ */
+class QueryCompatibleInputMiddlewareTest extends TestCase
```

Comment:
Should have a test assserting that non query compatible clients aren't having the handler added.